src/explorer/setAppWizardContextDefault.ts

```diff
@@ -9,32 +9,37 @@ import { Uri, workspace, WorkspaceConfiguration } from 'vscode';
 import { IAppServiceWizardContext, LinuxRuntimes, WebsiteOS } from 'vscode-azureappservice';
 import { LocationListStep } from 'vscode-azureextensionui';
 import { configurationSettings, extensionPrefix } from '../constants';
+import { IDeployWizardContext } from '../IDeployWizardContext';
 import { javaUtils } from '../utils/javaUtils';
 
-export async function setAppWizardContextDefault(wizardContext: IAppServiceWizardContext): Promise<void> {
-    const isJavaProject: boolean = await javaUtils.isJavaProject();
+export async function setAppWizardContextDefault(wizardContext: IAppServiceWizardContext & IDeployWizardContext): Promise<void> {
+    // if the user entered through "Deploy", we'll have a project to base our recommendations on
+    // otherwise, look at their current workspace and only suggest if one workspace is opened
+    const workspaceForRecommendation: string | undefined = wizardContext.fsPath ?
```

Comment:
Feels safer to get the containing workspace for fsPath and do your checks against that. Like [here](https://github.com/microsoft/vscode-azurefunctions/blob/master/src/utils/workspace.ts#L74)

Reply:
But if the project that they're deploying isn't opened in their workspace, this wouldn't work, right?  I thought we wanted to use the deployed project to set defaults regardless of whether or not it's in the current workspace.

Reply:
If you want to make it work in that scenario I'm totally fine with that. I was only thinking about the multi-root workspace when I originally filed the issue